app/src/main/java/org/mate/accessibility/AccessibilityViolation.java

```diff
@@ -9,19 +11,57 @@
     private Widget widget;
     private IScreenState state;
     private String info;
+    private boolean warning;
+
+    public void reportFlaw(int type, Widget widget, IScreenState state, String info){
+        if (widget==null){
+            widget = new Widget("artificialId","artificialClass","artificialIdByActivity");
+            widget.setX1(0);
+            widget.setY1(0);
+            widget.setX2(10);
+            widget.setY2(10);
+        }
+        String checkType = AccessibilityViolationTypes.NAMES[type];
+        String packageName = state.getPackageName();
+        String activityName = state.getActivityName();
+        String stateId = state.getId();
+        String widgetid = widget.getId();
+        widgetid = widgetid.replace(":","/");
+        info = info.replace(":","- ");
+        String widgetText = widget.getText();
+        String extraInfo = info;
+
+        String flawMsg = packageName+":"+activityName+":"+stateId+":"+checkType+":" + widget.getClazz() + ":" + widgetid + ":"+ widgetText;
```

Comment:
Not mandatory, but when creating a big string by concatenating many individual strings it is preferred to use a StringBuilder.

Reply:
done